### PR TITLE
Feature/#2

### PR DESCRIPTION
**Pull request by znikola**

### CODE REVIEW

File: src/cli/cli.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  if (printOutput) {
+    LogUtils.info(newContent);
+  }
+}
+
+// TODO: move to some other file?
+function replaceImports(content: string, startPosition: Position, endPosition: Position, newImports: string): string {
+  const splitted = content.split(NEW_LINE);
+  const deleteCount = endPosition.line - startPosition.line + 1;
+  splitted.splice(startPosition.line, deleteCount, ...newImports.split(NEW_LINE));
```
Comment (LTiger14) [inline src/cli/cli.ts:78]: Very nice!

File: src/lib/config/lib-config.model.ts [hunk 2]
```diff
@@ -1,9 +1,34 @@
...
+  printOutput = PRINT_OUTPUT_DEFAULT;
+  verbose = VERBOSE_DEFAULT;
+}
+
 /**
  * A config for sorting
  */
 export interface SortingConfig {
+  /** a file's content */
+  content: string;
```
Comment (LTiger14) [inline src/lib/config/lib-config.model.ts:31]: Why is this one mandatory?

File: src/lib/core/group-imports.ts [hunk 3]
```diff
@@ -1,19 +1,67 @@
 'use strict';
 
+import { SortingConfig } from '../config/lib-config.model';
 import { Import, ImportGroup, Type } from '../models/import';
 
-export function groupImports(imports: Import[]): ImportGroup[] {
+export function groupImports(imports: Import[], _config: SortingConfig): ImportGroup[] {
```
Comment (LTiger14) [inline src/lib/core/group-imports.ts:6]: Did you add the _config for later?

File: src/lib/core/group-imports.ts [hunk 4]
```diff
@@ -1,19 +1,67 @@
...
+
+  return groups;
+}
+
+function getRootFrom(from: string): string {
+  return from.split('/')[0];
+}
+
+function addToCurrent(groups: ImportGroup[], toAdd: Import): ImportGroup[] {
+  let lastGroup = groups[groups.length - 1];
```
Comment (LTiger14) [inline src/lib/core/group-imports.ts:60]: Good improvement!

File: src/lib/core/sorting.ts [hunk 5]
```diff
@@ -15,7 +16,7 @@ const INVALID_OR_ERROR = 0;
 
 const FOLDER_PATH_REGEX = new RegExp(FOLDER_PATH, 'g');
 
-export function sort(imports: Import[]): Import[] {
+export function sort(imports: Import[], _config: SortingConfig): Import[] {
```
Comment (LTiger14) [inline src/lib/core/sorting.ts:19]: _config again?

Comment (LTiger14) [reply src/lib/core/sorting.ts:19]: Oh I got it

File: src/lib/utils/file-utils.ts [hunk 6]
```diff
@@ -1,28 +1,54 @@
...
     return '';
   }
 
-  static saveFile(path: string, content: string): void {
-    if (this.isFile(path)) {
-      fs.writeFileSync(path, content, { encoding: UTF_8 });
+  static readDirectory(directoryPath: string, recursive: boolean, filePaths?: string[]): string[] {
+    filePaths = filePaths || [];
+    const files = fs.readdirSync(directoryPath);
+    for (const f in files) {
```
Comment (LTiger14) [inline src/lib/utils/file-utils.ts]: Why not for of and then directly use f instead of files[f]?

Comment (znikola) [reply src/lib/utils/file-utils.ts]: Good catch

File: src/lib/core/group-imports.ts (hunk 3 above)
Comment (znikola) [reply src/lib/core/group-imports.ts:6]: Yes, we might use it.

File: src/lib/config/lib-config.model.ts (hunk 2 above)
Comment (znikola) [reply src/lib/config/lib-config.model.ts:31]: Our CLI supports defining imports to sort inline, and for that we use `content`.
We also reuse it when iterating through `filePaths` [here](https://github.com/znikola/es6-typescript-import-sorter/pull/13/files#diff-c9ee77b8b3575c61b785ba838750243bR36) (`cli.ts`, line 36)

Do you think we can do this in a better way?

Comment (LTiger14) [reply src/lib/config/lib-config.model.ts:31]: Oh I get it. It's good, we can keep it like this I really don't have a better idea at the moment. 😛

**Review by LTiger14 — APPROVED** [review]

**Pull request merged**